fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -339,13 +349,61 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
 
         subscribeToTabCollections()
         updateLastBrowseActivity()
+        observeSaveToPDFErrorSnackBar()
+    }
+
+    @VisibleForTesting
+    internal fun observeSaveToPDFErrorSnackBar() {
```

Comment:
Before landing, please move code like this into their own classes so they are isolated and testable without growing large classes like the BrowserFragment one. There are examples like the "feature" component or the `AbstractBinding`-based classes.

If this is just for testing since the change is just a draft, please ignore this comment. 🙂